feature_engine/mathematical_transformers.py

```diff
@@ -0,0 +1,86 @@
+    The AdditionTransfomer() applies basic mathematical operations using aggregation of features.
+
+    Parameters
+    ----------
+
+    variables: list, default=None
+        The list of numerical variables to be transformed. If None, the transformer
+        will find and select all numerical variables.
+
+    math_operations: list, default=None
+        The list of basic math operations to be used in transformation.
+
+        Each operation should be a string and must be one of elements
+        from list: ['sum', 'prod', 'mean', 'std', 'max', 'min']
+
+        Each operation will result in operation column in result dataset.
+
+        If None, the transformer will calculate each of permitted
+        operations: ['sum', 'prod', 'mean', 'std', 'max', 'min']
+    """
+
+    def __init__(self, variables=None, math_operations=None):
+        self.variables = variables
+        self.math_operations = math_operations
+        self.math_operations_permitted = ['sum', 'prod', 'mean', 'std', 'max', 'min']
+
+        if self.math_operations is None:
+            self.operations_ = self.math_operations_permitted
+
+        elif isinstance(self.math_operations, list):
+            if any(elem_par not in self.math_operations_permitted for elem_par in self.math_operations):
+                raise KeyError("At least one of math_operations is not found in permitted operations set. "
+                               "Choose one of ['sum', 'prod', 'mean', 'std', 'max', 'min']")
+            self.operations_ = self.math_operations
+        else:
+            raise KeyError("math_operations parameter must be a list or None")
+
+    def fit(self, X, y=None):
+        """
+        Fits source dataset, verifies if variables parameter contains more than one variable.
+
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+            Can be the entire dataframe, not just the variables to transform.
+
+        y : None
+            y is not needed in this transformer. You can pass y or None.
+        """
+        X = super().fit(X, y)
+        self.input_shape_ = X.shape
+
+        if len(self.variables) <= 1:
+            raise KeyError("AdditionTransformer requires two or more features to make proper transformations.")
+
+        return self
+
+    def transform(self, X):
+        """
+        Transforms source dataset.
+
+        Adds column for each operation with calculation based on variables and operation.
+
+        Parameters
+        ----------
+
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The data to transform.
+
+        Returns
+        -------
+
+        X_transformed : pandas dataframe of shape = [n_samples, n_features + n_operations]
+            The dataframe with operations results added.
+        """
+        X = super().transform(X)
+
+        for operation in self.operations_:
+            variables_set_name = f"{operation}({','.join(self.variables)})"
```

Comment:
is this returning the operation plus all the columns separated by a comma? I wonder if this is a good idea. I think for cases where the user wants to sum variables indiscriminately and without really giving a lot of thought on what they are doing it make work. But in my experience, when we add variables, we add variables that make sense to add. For example, if a car had an accident, and we have binary variables indicating which part of the car is broken, we add those up into a variable called "total_damage"

I think I would prefer giving the user the posibility to name their variables. Could we add this in the init? and explain it in the docstrings?

Reply:
Yes, exactly as you described - operation plus all columns separated by comma. 
It's funny, I had this in my intermediate version of code, but decided not to allow user to do this. If user had two MathematicalVariableCombinator with other columns set but same "alias", the second result set would overwrite the first...

But it seems it's better to allow user to use such alias, I'll implement this in next commit.

Reply:
just one thing to verify - if there are more than one operation, we need to reflect this in results, so I'll implement this that if there are more than one operations to calculate, I'll leave operation in result column.